src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -24,32 +24,225 @@
 
 package org.takes.facets.auth.social;
 
+import java.io.FileInputStream;
 import java.io.IOException;
-import org.junit.Ignore;
+import java.net.HttpURLConnection;
+import javax.json.Json;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
 import org.junit.Test;
+import org.mockito.Mockito;
+import org.takes.Response;
+import org.takes.facets.auth.Identity;
+import org.takes.misc.Href;
+import org.takes.rq.RqFake;
+import org.takes.rs.RsWithBody;
+import org.takes.rs.RsWithStatus;
+import org.takes.rs.RsWithType;
 
 /**
  * Test case for {@link PsLinkedin}.
  * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
  * @version $Id$
  * @since 0.11.3
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
  */
 public final class PsLinkedinTest {
+    /**
+     * Correct fake REQUEST.
+     */
+    private static final RqFake REQUEST = new RqFake("GET", "/?code=200");
 
     /**
      * PsLinkedin can login.
      * @throws IOException If some problem inside
-     * @todo #12:30min/DEV This class not implemented yet, but has to be.
-     *  Please implement it. Mocked OAuth server is one of the possible
-     *  solutions. In this case, we could use some library like WireMock
-     *  (see http://www.wiremock.org), which allows make stub for web
-     *  service. We'll need to make some changes in PsLinkedin,
-     *  that allow to change API url, and use local stubbed http
-     *  server during unit test.
-     */
-    @Ignore
+     */
+    @Test
+    public void entersSuccessfully() throws IOException {
+        final MemberProfileJson memberprofilejson =
+            Mockito.mock(MemberProfileJson.class);
+        Mockito.when(
+            memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUser.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        MatcherAssert.assertThat(
+            new PsLinkedin(memberprofilejson)
+                .enter(REQUEST).next().urn(),
+            Matchers.is("urn:linkedin:1")
+        );
+    }
+
+    /**
+     * Login fails if authorization code is not provided with the REQUEST.
```

Comment:
@ikhvostenkov Now javadocs and method names are much better, but you can make it even better :) , please see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names